core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -332,6 +332,8 @@ private TableProperties() {}
   public static final String MERGE_MODE = "write.merge.mode";
   public static final String MERGE_MODE_DEFAULT = "copy-on-write";
 
+  public static final String COPY_ON_WRITE_MODE = "copy-on-write";
+  public static final String MERGE_ON_READ_MODE = "merge-on-read";
```

Comment:
I suggest we use the name `MERGE_MODE_COPY_ON_WRITE`\`MERGE_MODE_MERGE_ON_READ_MODE` , delete `MERGE_MODE_DEFAULT` at the same time.

Reply:
MERGE_MODE_COPY_ON_WRITE would indicate that this is used for merge mode, but actually it's also for delete and update mode.

I wouldn't delete MERGE_MODE_DEFAULT because in my opinion that is for different purpose than the ones I introduced here. Let's say that someone wants to change the default from "copy-on-write" to "merge-on-read". Then only this DEFAULT variable has to be changed. However, with your suggestion one has to search for the uses of COPY_ON_WRITE_MODE (if we stick with the original name) and replace them with MERGE_ON_READ_MODE, but not all of them only the ones where the default is required.

Reply:
Yes, I did overlook that. I initially suggested a different name because adding them individually to TableProperties felt a little weird.